common/kernel.c

```diff
@@ -63,6 +66,13 @@ void kernel_main(void) {
 
     printk("All tasks done.\n");
 
-    while (1)
-        halt();
+    keyboard_setup();
+
+    while (1) {
+        io_delay();
+        key = keyboard_next_key();
+        if (isprint(key)) {
+            printk("%c", key);
```

Comment:
Better use putchar() here

Reply:
`putchar` outputs to the serial console, printk works also on vga